mono/mini/simd-intrinsics-netcore.c

```diff
@@ -625,25 +658,219 @@ static guint16 bmi2_methods [] = {
 	SN_get_IsSupported,
 };
 
+static gboolean
+is_intrinsics_class (MonoClass *klass, const char *name, gboolean *is_64bit)
+{
+	const char *class_name = m_class_get_name (klass);
+	if ((!strcmp (class_name, "X64") || !strcmp (class_name, "Arm64")) && klass->nested_in) {
+		*is_64bit = TRUE;
+		return !strcmp (m_class_get_name (klass->nested_in), name); 
+	}
+	else {
+		*is_64bit = FALSE;
+		return !strcmp (class_name, name);
+	}
+}
+
+static MonoTypeEnum 
+get_vector_underlying_type (MonoType *vector_type)
+{
+	MonoClass* klass = mono_class_from_mono_type_internal (vector_type);
+	MonoType* underlying_type = mono_class_get_context (klass)->class_inst->type_argv [0];
+	return underlying_type->type;
+}
+
 static MonoInst*
 emit_x86_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **args)
 {
-	const char *class_name;
-	const char *class_ns;
 	MonoInst *ins;
 	int id;
 	gboolean supported, is_64bit;
 	MonoClass *klass = cmethod->klass;
 
-	class_ns = m_class_get_name_space (klass);
-	class_name = m_class_get_name (klass);
-	if (!strcmp (class_name, "Popcnt") || (!strcmp (class_name, "X64") && cmethod->klass->nested_in && !strcmp (m_class_get_name (cmethod->klass->nested_in), "Popcnt"))) {
+	if (is_intrinsics_class (klass, "Sse", &is_64bit)) {
+		id = lookup_intrins (sse_methods, sizeof (sse_methods), cmethod);
+		if (id == -1)
+			return NULL;
+
+		supported = COMPILE_LLVM (cfg) && (mini_get_cpu_features (cfg) & MONO_CPU_X86_SSE) != 0 &&
+			m_class_get_image (cfg->method->klass) != mono_get_corlib (); // We only support the subset used by corelib
+		
+		switch (id) {
+		case SN_get_IsSupported:
+			EMIT_NEW_ICONST (cfg, ins, 0); // TODO: implement subset used by corlib
+			ins->type = STACK_I4;
```

Comment:
Will corlib still work if this returns 0 ?

Reply:
Eventually will enable this but there are cases when Sse/Sse2 are used under `if (Sse41.IsSupported)` which is already implemented for corlib.